### PR TITLE
Refactor Jenkinsfile

### PR DESCRIPTION
**Pull request by cobylu**
- Moved functions to a groovy file
- Updated setProperty to work with commented or non-commented
properties
- Made formatting consistent on using parens around user-
defined functions

### CODE REVIEW

**Comment by cobylu** [comment]
ready for cr @hankjtaylor @samratjha96

**Comment by samratjha96** [comment]
Since we haven't fully gotten a sign off on whether this or the scripted pipeline is the way we want to go, I think this pull request should stay active. I would add another commit that shows the other way so everything's available to look at and decide on.

File: groovy/JenkinsUtils.groovy [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  sh "cp devops/adm/" + versionPropertyFile + " adm/appian-version-client/version-manager.properties"
+  dir("adm/appian-version-client") {
+    setProperty("version-manager.properties", "vcUsername", "${REPOUSERNAME}")
+    setProperty("version-manager.properties", "vcPassword", "${REPOPASSWORD}")
+    setProperty("version-manager.properties", "appianObjectsRepoPath", "appian/applications/${APPLICATIONNAME}")
+    sh "./version-application.sh -package_path ../app-package.zip"
+  }
+}
+
+
```
Comment (samratjha96) [inline groovy/JenkinsUtils.groovy]: Remove line

**Comment by samratjha96** [comment]
LGTM 👐

File: Jenkinsfile [hunk 2]
```diff
@@ -3,66 +3,97 @@ pipeline {
...
+        script {
+          def jenkinsUtils = load "groovy/JenkinsUtils.groovy"
+          jenkinsUtils.runTests("fitnesse-automation.integrate.properties")
+        }
       }
       post {
         always { dir("f4a/FitNesseForAppian"){ junit "fitnesse-results.xml" } }
-        failure { archiveArtifacts artifacts: retrieveLogs("fitnesse-automation.integrate.properties"), fingerprint: true }
+        failure {
+          script { def jenkinsUtils = load "groovy/JenkinsUtils.groovy" archiveArtifacts artifacts: jenkinsUtils.retrieveLogs("fitnesse-automation.integrate.properties"), fingerprint: true }
```
Comment (hankjtaylor) [inline Jenkinsfile]: although it lengthens the file, may want to make this multi-lined. It was clean when we were just calling one command, but now that we define the utility before, it gets lengthier

File: Jenkinsfile [hunk 3]
```diff
@@ -3,66 +3,97 @@ pipeline {
...
+        script {
+          def jenkinsUtils = load "groovy/JenkinsUtils.groovy"
+          jenkinsUtils.runTests("fitnesse-automation.acceptance.properties")
+        }
       }
       post {
         always { dir("f4a/FitNesseForAppian"){ junit "fitnesse-results.xml" } }
-        failure { archiveArtifacts artifacts: retrieveLogs("fitnesse-automation.acceptance.properties"), fingerprint: true }
+        failure {
+          script { def jenkinsUtils = load "groovy/JenkinsUtils.groovy" archiveArtifacts artifacts: retrieveLogs("fitnesse-automation.acceptance.properties"), fingerprint: true }
```
Comment (hankjtaylor) [inline Jenkinsfile]: same commend as above for this line

**Pull request merged**